.github/workflows/build-ruby-generator.yml

```diff
@@ -1,6 +1,6 @@
 name: Build Ruby Generator
 
-on: push
+on: pull_request
```

Comment:
`pull_request` will build on every push to PRs, so it'll still build on a push but only when there's an associated PR.